client/src/routes/settings/+page.svelte

```diff
@@ -1,31 +1,198 @@
 <script lang="ts">
-  import { deleteAccount, logOut } from "$lib/firebase/auth";
+  import {
+    deleteAccount,
+    linkWithGoogle,
+    linkWithMicrosoft,
+    linkWithPassword,
+    logOut,
+    userHasGoogleProvider,
+    userHasMicrosoftProvider,
+  } from "$lib/firebase/auth";
   import { goto } from "$app/navigation";
   import Dialog, { Title, Content, Actions } from "@smui/dialog";
   import Button, { Label } from "@smui/button";
+  import { Svg, Icon } from "@smui/common";
+  import IconButton from "@smui/icon-button";
+  import Textfield from "@smui/textfield";
+  import { authStore as user } from "$stores/auth";
 
   let showDeletionPrompt = false;
-  let errorMsg = "";
+  let showOptions = false;
+  let showPassword = false;
   let errPrompt = false;
+  let linkPass = false;
+
+  let password = "";
+  let errorMsg = "";
+  let googleErr = "";
+  let microsoftErr = "";
+  let passErr = "";
+  let deleteErr = "";
 
+  function outputErrMsg(error: string) {
+    switch (error) {
+      case "Firebase: Password should be at least 6 characters (auth/weak-password).":
+        return "Password should be at least 6 characters";
+      case "google-account-already-linked":
+        return "google account already linked";
+      case "microsoft-account-already-linked":
+        return "microsoft account already linked";
+      default:
+        return "An unexpected error has occured";
+    }
+  }
   function verifyDelete() {
     try {
       deleteAccount();
       // If no error
-      errorMsg = "";
-
       goto("/");
-      return;
     } catch (err) {
       errPrompt = true;
-      errorMsg = err instanceof Error ? err.message : String(err);
+      deleteErr = err instanceof Error ? err.message : String(err);
+      outputErrMsg(deleteErr);
+    }
+  }
+
+  async function linkGoogleAccount() {
+    try {
+      await linkWithGoogle($user);
+      window.location.reload();
+    } catch (err) {
+      googleErr = err instanceof Error ? err.message : String(err);
+      outputErrMsg(googleErr);
+    }
+  }
+
+  async function linkMicrosoftAccount() {
+    try {
+      await linkWithMicrosoft($user);
+      window.location.reload();
+    } catch (err) {
+      microsoftErr = err instanceof Error ? err.message : String(err);
+      outputErrMsg(microsoftErr);
+    }
+  }
+
+  async function linkPassword() {
+    try {
+      await linkWithPassword(password);
+      linkPass = true;
+      showOptions = false;
+      clearFields();
+    } catch (err) {
+      passErr = err instanceof Error ? err.message : String(err);
     }
   }
+
+  function clearFields() {
+    password = "";
+  }
+
+  const googleSvgPaths: { color: string; path: string }[] = [
```

Comment:
duplicate code from signinbutton component. make another component and insert them in each